src/RegionCosts.cpp

```diff
@@ -0,0 +1,622 @@
+    if (size == unknown) {
+        return unknown;
+    }
+    int64_t size_per_ele = get_func_value_size(f);
+    return size * size_per_ele;
+}
+
+int64_t RegionCosts::region_footprint(const map<string, Box> &regions,
+                                      const set<string> &inlined) {
+    map<string, int> num_consumers;
+    for (const auto &f : regions) {
+        num_consumers[f.first] = 0;
+    }
+    for (const auto &f : regions) {
+        map<string, Function> prods = find_direct_calls(get_element(env, f.first));
+        for (const auto &p : prods) {
+            auto iter = num_consumers.find(p.first);
+            if (iter != num_consumers.end()) {
+                iter->second += 1;
+            }
+        }
+    }
+
+    vector<Function> outs;
+    for (const auto &f : num_consumers) {
+        if (f.second == 0) {
+            outs.push_back(get_element(env, f.first));
+        }
+    }
+
+    // Realization order
+    vector<string> order = realization_order(outs, env);
+
+    int64_t working_set_size = 0;
+    int64_t curr_size = 0;
+
+    map<string, int64_t> func_sizes;
+
+    for (const auto &f : regions) {
+        // Inlined functions do not have allocations
+        bool is_inlined = inlined.find(f.first) != inlined.end();
+        int64_t size = is_inlined ? 0 : region_size(f.first, f.second);
+        if (size == unknown) {
+            return unknown;
+        } else {
+            func_sizes.emplace(f.first, size);
+        }
+    }
+
+    for (const auto &f : order) {
+        if (regions.find(f) != regions.end()) {
+            curr_size += get_element(func_sizes, f);
+        }
+        working_set_size = std::max(curr_size, working_set_size);
+        map<string, Function> prods = find_direct_calls(get_element(env, f));
+        for (const auto &p : prods) {
+            auto iter = num_consumers.find(p.first);
+            if (iter != num_consumers.end()) {
+                iter->second -= 1;
+                if (iter->second == 0) {
+                    curr_size -= get_element(func_sizes, p.first);
+                    internal_assert(curr_size >= 0);
+                }
+            }
+        }
+    }
+
+    return working_set_size;
+}
+
+int64_t RegionCosts::input_region_size(string input, const Box &region) {
+    int64_t size = box_size(region);
+    if (size == unknown) {
+        return unknown;
+    }
+    int64_t size_per_ele = get_element(inputs, input).bytes();
+    return size * size_per_ele;
+}
+
+int64_t RegionCosts::input_region_size(const map<string, Box> &input_regions) {
+    int64_t total_size = 0;
+    for (const auto &reg : input_regions) {
+        int64_t size = input_region_size(reg.first, reg.second);
+        if (size == unknown) {
+            return unknown;
+        } else {
+            total_size += size;
+        }
+    }
+    return total_size;
+}
+
+void RegionCosts::disp_func_costs() {
+    debug(3) << "===========================" << '\n';
```

Comment:
I think that functions with the explicit purpose of printing something out should do so unconditionally, otherwise it's a bit confusing to have to both use this function *and* set the debug level to 3 or more.